### PR TITLE
energy mng demo: settings menu

### PR DESCRIPTION
**Pull request by FloVanGH**
* SettingsOverview
* About
* Settings page
* IconButton
* ScrollView
* ListView
* Item / ItemGroup
* CheckBox
* RadioButton
* Switch

### CODE REVIEW

File: examples/energy_mng/ui/widgets/scroll_view.slint [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    
+    min-height: 50px;
+    min-width: 50px;
+    horizontal-stretch: 1;
+    vertical-stretch: 1;
+
+    i-flickable := Flickable {
+        x: 2px;
+        y: 2px;
+        interactive: false;
```
Comment (ogoffart) [inline examples/energy_mng/ui/widgets/scroll_view.slint:68]: On embedded, we want this to be interactive

File: examples/energy_mng/ui/pages/settings.slint [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            }
+
+            i-about-button := IconButton { 
+                icon: @image-url("../assets/information.svg");
+            }
+        }
+
+        ListView {  
+            model: root.model;
+            selected-index <=> root.selected-index;
```
Comment (ogoffart) [inline examples/energy_mng/ui/pages/settings.slint:283]: When something is selected here, it should hide the menu and go to that page.

File: examples/energy_mng/ui/pages/settings.slint [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        { text: "Usage"},
+        { text: "Heater"},
+        { text: "Weather"},
+        { text: "Balance"},
+        { text: "Common"},
+    ];
+    in-out property <int> selected-index;
+}
+
+component SettingsOverview inherits Page {
```
Comment (ogoffart) [inline examples/energy_mng/ui/pages/settings.slint:250]: This is not really a SettingsOverview,  this is the menu to go from page to page

File: examples/energy_mng/ui/pages/settings.slint [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+        // spacer
+        ScrollView {  
+            vertical-stretch: 1;
+
+            VerticalLayout {  
+                alignment: start;
+                spacing: Theme.spaces.medium;
+
+                Item {
```
Comment (ogoffart) [inline examples/energy_mng/ui/pages/settings.slint:83]: In this part, i wonder if clicking on the whole item shouldn't toggle it.

File: examples/energy_mng/ui/widgets/scroll_view.slint (hunk 1 above)
Comment (FloVanGH) [reply examples/energy_mng/ui/widgets/scroll_view.slint:68]: done

File: examples/energy_mng/ui/pages/settings.slint (hunk 3 above)
Comment (FloVanGH) [reply examples/energy_mng/ui/pages/settings.slint:250]: done

File: examples/energy_mng/ui/pages/menu_page/settings.slint [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        padding-bottom: Theme.spaces.large;
+        spacing: Theme.spaces.medium;
+
+        HorizontalLayout {  
+            vertical-stretch: 0;
+            alignment: start;
+
+            i-back-button := IconButton {
+                horizontal-stretch: 0;
+                icon: @image-url("../assets/arrow-left.svg");
```
Comment (ogoffart) [inline examples/energy_mng/ui/pages/menu_page/settings.slint:71]: I noticed the following warning:

```
warning: Cannot find image file ../assets/arrow-left.svg
warning:   --> /home/olivier/slint/examples/energy_mng/ui/pages/menu_page/settings.slint:71:23
warning:    |
warning: 71 |                 icon: @image-url("../assets/arrow-left.svg");
warning:    |                       ^
```


I guess it should be:

```suggestion
                icon: @image-url("../../assets/arrow-left.svg");
```

**Review by ogoffart — APPROVED** [review]

**Pull request merged**